extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonSerializerFactory.java

```diff
@@ -76,29 +86,45 @@
  *
  * <pre>{@code
  * public class Person$quarkusjacksonserializer extends StdSerializer {
+ *     static final String[] address_ROLES_ALLOWED = new String[] {"admin"};
+ *
  *     public Person$quarkusjacksonserializer() {
  *         super(Person.class);
  *     }
  *
  *     public void serialize(Object var1, JsonGenerator var2, SerializerProvider var3) throws IOException {
  *         Person var4 = (Person) var1;
  *         var2.writeStartObject();
+ *         var2.writeFieldName(SerializedStrings$quarkusjacksonserializer.age);
  *         int var5 = var4.getAge();
- *         var2.writeNumberField("age", var5);
+ *         var2.writeNumber(var5);
+ *         var2.writeFieldName(SerializedStrings$quarkusjacksonserializer.firstName);
  *         String var6 = var4.getFirstName();
- *         var2.writeStringField("firstName", var6);
+ *         var2.writeString("firstName", var6);
+ *         var2.writeFieldName(SerializedStrings$quarkusjacksonserializer.familyName);
  *         String var7 = var4.getLastName();
- *         var2.writeStringField("lastName", var7);
- *         String[] var8 = new String[] { "admin" };
- *         Address var9 = var4.getAddress();
- *         if (JacksonMapperUtil.includeSecureField(var8)) {
- *             var2.writePOJOField("address", var9);
+ *         var2.writeString(var7);
+ *         if (JacksonMapperUtil.includeSecureField(address_ROLES_ALLOWED)) {
+ *             var2.writeFieldName(SerializedStrings$quarkusjacksonserializer.address);
+ *             Address var9 = var4.getAddress();
+ *             var2.writePOJO(var9);
  *         }
  *         var2.writeEndObject();
  *     }
  * }
+ *
+ * public class SerializedStrings$quarkusjacksonserializer {
```

Comment:
@mariofusco why having a separate class for this? Is it meant to be used by others?

Reply:
Yes, that class is shared by all the pojos in the same package.

Reply:
The class is separate because it contains all `SerializedString` instances collected for all types than can be serialized - and some of these fields could be shared by different types.